src/gc.c

```diff
@@ -3432,10 +3417,7 @@ JL_DLLEXPORT void jl_gc_collect(jl_gc_collection_t collection)
     jl_task_t *ct = jl_current_task;
     jl_ptls_t ptls = ct->ptls;
     if (jl_atomic_load_acquire(&jl_gc_disable_counter)) {
-        size_t localbytes = jl_atomic_load_relaxed(&ptls->gc_num.allocd) + gc_num.interval;
-        jl_atomic_store_relaxed(&ptls->gc_num.allocd, -(int64_t)gc_num.interval);
-        static_assert(sizeof(_Atomic(uint64_t)) == sizeof(gc_num.deferred_alloc), "");
-        jl_atomic_fetch_add((_Atomic(uint64_t)*)&gc_num.deferred_alloc, localbytes);
```

Comment:
Are we sure we don't need anything like this anymore? When gc is disabled, we will start hitting a point where `jl_gc_collect` is called and `jl_gc_disable_counter` is atomically loaded on every allocation. Is that ok?

Reply:
So this is one of the things I'm not super sure. One thing that we could do here is to set the next collection to a very high value so we don't try to enter it at all. And then forcibly run a gc when we reanable it?

Reply:
The other is to update it with a simple default algorithm so that it does trigger but not so often.